### DISCUSSION TITLE
Usage of Game Variables

### DISCUSSION THREAD

**Discussion by worldpeace-germany** [missionpinball/missionpinball#117] [Q&A] [answered]
Hello,

this question is about [_game_ variables](https://missionpinball.org/game_vars/), it is not about player or machine variables. Two simple questions:

a) Can I create my own game variables (like I can create my own machine and player variables)?
If yes, how? If no, is there a specific reason or is that just a gap in the software?

b) Can I display a game variable (own if a)=yes and/or build-in game variable) on a slide?
If yes, how? If no, is there a specific reason or is that just a gap in the software?

Thanks!

**Comment by atummons** [comment, marked as answer]
A) Yes you can, but not like player and machine. Game is a specific mode that handles the Game mode of the machine. It is built to track the data and flow of a game. If you want to track extra variables by game you can modify the game code file and add your own. This would be via Python.

Alternatively, use machine variables and just have a handler at game_ended/game_started (or whatever event you choose) to set the machine_var back to 0 (or your start value).

B) As far as slides everything can be called and used depending how you move forward based on the above.

**Reply by toomanybrians** [reply]
I agree with @atummons about using machine variables. In fact that's the intended design! (The idea being that a machine variable is machine-wide which is what game variables would be.) For your use case, set them up with `persist: false` so they don't persist to disk, and setup a [`variable_player:`](https://missionpinball.org/config/variable_player/#action) (with `action: set_machine`) that sets them all to zero or (whatever initial state you want) on one of the game start events.

**Comment by worldpeace-germany** [comment]
Thanks for the quick response. In regards to 

a)
Perfectly fine I can live with that, from the documentation it just wasn't clear to me. I have created a pull request to update the documentation so that nobody has to ask the same question again :-) https://github.com/missionpinball/mpf-docs/pull/490/commits/1f03ba1d6e525fc77d008c8353959ea1a24d31e5

b)
@atummons You wrote "Everything can be called". That sounds good. Concrete question: How can I display this variable https://missionpinball.org/game_vars/num_players/ on a slide?

**Comment by worldpeace-germany** [comment]
very old and the last question on the slide is not valid anymore since Godot is being used. So I am closing this discussion.
